### PR TITLE
Https enforcement

### PR DESCRIPTION
**Pull request by vibe13**
Added enforcement of https. 
@lfryc @abstractj @matzew @bparees  mind to check? Thanks!

### CODE REVIEW

**Comment by lfryc** [comment]
testing :eyes:

File: versions/1.0.2/standalone/configuration/standalone.xml [hunk 1]
```diff
@@ -553,7 +553,7 @@
     </interfaces>
 
     <socket-binding-group name="standard-sockets" default-interface="public" port-offset="0">
-        <socket-binding name="http" port="8080"/>
+        <socket-binding name="https" port="${jboss.https.port:8443}"/>
```
Comment (bparees) [inline versions/1.0.2/standalone/configuration/standalone.xml:556]: I don't think this really matters since we terminate SSL at the proxy.

**Comment by bparees** [comment]
looks ok to me.

**Comment by abstractj** [comment]
@vibe13 Atterrato!

**Comment by lfryc** [comment]
Sorry guys, I wasn''t able to test it on OpenShift as 

```
rhc app create --no-git -g medium upshttpsenforce4 https://raw.githubusercontent.com/lfryc/openshift-origin-cartridge-aerogear-push/vibe13-https_enforcement/metadata/manifest.yml
```

always fail, I thought it might be just temporary issue but it doesn't as of now as well:

```
Creating application 'upshttpsenforce4' ... C
Starting aerogear-push cart
Timed out waiting for http listening port
Failed to execute: 'control start' for /var/lib/openshift/5464cefdecb8d4436f0000c9/aerogear-push
```

That's expected I would say, but I don't know what we can do to work around it.

**Comment by abstractj** [comment]
@lfryc Hmmmm, I'm getting this error now

```
The cartridge manifest at 'https://github.com/aerogear/openshift-origin-cartridge-aerogear-push/blob/master/metadata/manifest.yml' could not be downloaded: The cartridge manifest
at 'https://github.com/aerogear/openshift-origin-cartridge-aerogear-push/blob/master/metadata/manifest.yml' must be smaller than 20480 bytes.
```

It didn't happen before, my cartridge is not medium size, just play small gears.

**Comment by bparees** [comment]
@abstractj  you need to point to the raw manifest: https://raw.githubusercontent.com/aerogear/openshift-origin-cartridge-aerogear-push/master/metadata/manifest.yml

**Comment by abstractj** [comment]
@bparees thank you! Now I got another error:

```
Creating application 'goose' ...
The cartridge manifest at 'https://github.com/aerogear/openshift-origin-cartridge-aerogear-push/blob/master/metadata/manifest.yml' could not be downloaded: The cartridge manifest
at 'https://github.com/aerogear/openshift-origin-cartridge-aerogear-push/blob/master/metadata/manifest.yml' must be smaller than 20480 bytes.
~ ❯❯❯ rhc app create --no-git goose https://raw.githubusercontent.com/aerogear/openshift-origin-cartridge-aerogear-push/master/metadata/manifest.yml                              ⏎
The cartridge 'https://raw.githubusercontent.com/aerogear/openshift-origin-cartridge-aerogear-push/master/metadata/manifest.yml' will be downloaded and installed

Application Options
-------------------
Domain:     abstractj
Cartridges: https://raw.githubusercontent.com/aerogear/openshift-origin-cartridge-aerogear-push/master/metadata/manifest.yml
Gear Size:  default
Scaling:    no

Creating application 'goose' ...
The provided downloadable cartridge 'https://raw.githubusercontent.com/aerogear/openshift-origin-cartridge-aerogear-push/master/metadata/manifest.yml' cannot be loaded: Source-Url
is required in manifest to obtain cartridge via URL
```

**Comment by bparees** [comment]
@abstractj what exactly are you trying to do?  if you want to run the normal aerogear cartridge, you need to bounce the manifest through the cart reflector:
rhc app create --no-git yourapp https://cartreflect-claytondev.rhcloud.com/reflect?github=aerogear/openshift-origin-cartridge-aerogear-push

which will add the source-url.

if you're trying to run this specific branch, you can point here: https://raw.githubusercontent.com/lfryc/openshift-origin-cartridge-aerogear-push/vibe13-https_enforcement/metadata/manifest.yml

because that manifest.yml contains a source-url entry.

**Comment by bparees** [comment]
turns out this PR broke the cart, sorry I overlooked that you had removed the http port from the configuration...that was not a valid change.  Fix is here:
https://github.com/aerogear/openshift-origin-cartridge-aerogear-push/pull/29

**Comment by lfryc** [comment]
Thanks @bparees that fixed it!

**Pull request merged**